DIMS.nf

```diff
@@ -0,0 +1,136 @@
+include { UnidentifiedFillMissing } from './CustomModules/DIMS/UnidentifiedFillMissing.nf' params(scripts_dir:"$params.scripts_dir", thresh:"$params.thresh", resolution:"$params.resolution", ppm:"$params.ppm")
+include { UnidentifiedCalcZscores } from './CustomModules/DIMS/UnidentifiedCalcZscores.nf' params(scripts_dir:"$params.scripts_dir", ppm:"$params.ppm", zscore:"$params.zscore")
+
+
+// define parameters
+// def samplesheet     = params.samplesheet
+// def nr_replicates   = params.nr_replicates
+def raw_files       = extractRawfilesFromDir(params.rawfiles_path)
+def analysis_id     = params.outdir.split('/')[-1]
+// def resolution      = params.resolution
+// def hmdb_db_file    = params.hmdb_db_file
+// def ppm             = params.ppm
+// def standard_run    = params.standard_run
+// def scripts_dir     = params.scripts_dir
+// def thresh          = params.thresh
+// def zscore          = params.zscore
+// def matrix          = params.matrix
+// def relevance_file  = params.relevance_file
+
+
+workflow {
+    // create init.RData file with info on technical replicates
+    MakeInit(tuple(params.samplesheet, params.nr_replicates))
+
+    // Read raw files and convert to mzML format
+    ConvertRawFile(raw_files)
+    
+    // Generate breaks on one of the mzML files
+    GenerateBreaks(ConvertRawFile.out.take(1))
+
+    // Generate HMDB parts for parallel processing in SumAdducts step
+    // HMDB without adducts, without isotopes, only main entry for each metabolite
+    HMDBparts_main(params.hmdb_db_file, GenerateBreaks.out)
+
+    // Generate HMDB parts for parallel processing in PeakGrouping step
+    HMDBparts(params.hmdb_db_file, GenerateBreaks.out)
+
+    // Assign intensities to bins (breaks) per mzML file
+    AssignToBins(ConvertRawFile.out.combine(GenerateBreaks.out))
+
+    // Average intensities over technical replicates for each sample
+    AverageTechReplicates(AssignToBins.out.collect(), MakeInit.out)
+
+    // Peak finding per sample
+    PeakFinding(AverageTechReplicates.out.binned.collect().flatten().combine(GenerateBreaks.out))
+
+    // Spectrum peak finding per sample
+    SpectrumPeakFinding(PeakFinding.out.collect(), AverageTechReplicates.out.patterns)
+
+    // Peak grouping over samples: identified part
+    PeakGrouping(HMDBparts.out.collect().flatten(), SpectrumPeakFinding.out, AverageTechReplicates.out.patterns)
+
+    // Fill missing values in peak group list: identified part
+    FillMissing(PeakGrouping.out.grouped_identified, AverageTechReplicates.out.patterns)
+
+    // Collect filled peak group list: identified part
+    CollectFilled(FillMissing.out.collect(), AverageTechReplicates.out.patterns)
+
+    // Sum adducts of each metabolite per scan mode: identfied part
+    SumAdducts(CollectFilled.out.filled_pgrlist, AverageTechReplicates.out.patterns, HMDBparts_main.out.collect().flatten())
```

Comment:
Restyle -> divide on multiple lines.

Reply:
Done for each process with more than one input channel